tools/interactive/interactivetool_ml_jupyter_notebook.xml

```diff
@@ -0,0 +1,94 @@
+        </entry_point>
+    </entry_points>
+    <environment_variables>
+        <environment_variable name="HISTORY_ID">$__history_id__</environment_variable>
+        <environment_variable name="REMOTE_HOST">$__galaxy_url__</environment_variable>
+        <environment_variable name="GALAXY_WEB_PORT">8080</environment_variable>
+        <environment_variable name="GALAXY_URL">$__galaxy_url__</environment_variable>
+        <environment_variable name="DISABLE_AUTH">true</environment_variable>
+        <environment_variable name="API_KEY" inject="api_key" />
+    </environment_variables>
+    <command detect_errors="aggressive"><![CDATA[
+        #import re
+        export GALAXY_WORKING_DIR=`pwd` &&
+        mkdir -p ./jupyter/outputs/ &&
+        mkdir -p ./jupyter/data &&
+        mkdir -p ./jupyter/notebooks &&
+        mkdir -p ./jupyter/elyra &&
+
+        #set $cleaned_name = re.sub('[^\w\-\.]', '_', str($input.element_identifier))
+        ln -sf '$input' './jupyter/data/${cleaned_name}' &&
+
+        ## change into the directory where the notebooks are located
+        cd ./jupyter/ &&
+        export HOME=/home/\$NB_USER/ &&
+        export PATH=/home/\$NB_USER/.local/bin:\$PATH &&
+        #if $mode.mode_select == 'scratch':
+            ## copy all notebooks, workflows and data
+            cp /home/\$NB_USER/home_page.ipynb ./ &&
+            cp /home/\$NB_USER/notebooks/*.ipynb ./notebooks/ &&
+            cp /home/\$NB_USER/elyra/*.* ./elyra/ &&
+            cp /home/\$NB_USER/data/*.tsv ./data/ &&
+            ## provide all rights to copied files
+            chown -R \$NB_USER:users `pwd` &&
+            chown \$NB_USER ./home_page.ipynb &&
+            chown \$NB_USER ./notebooks/ &&
+            chown \$NB_USER ./notebooks/*.ipynb &&
+            chown \$NB_USER ./elyra/ &&
+            chown \$NB_USER ./elyra/*.* &&
+            jupyter trust ./home_page.ipynb &&
+            jupyter trust ./notebooks/*.ipynb &&
+            jupyter trust ./elyra/*.ipynb &&
+            jupyter lab --allow-root --no-browser --NotebookApp.shutdown_button=True
+	#end if
+    ]]>
+    </command>
+    <inputs>
+
+        <conditional name="mode">
+            <param name="mode_select" type="select" label="Do you already have a notebook?" help="If not, no problem we will provide you with a default one.">
+                <option value="scratch">Start with a fresh notebook</option>
+                <option value="previous">Load a previous notebook</option>
+            </param>
+            <when value="scratch"/>
+            <when value="previous">
+                <param name="ipynb" type="data" format="ipynb" label="IPython Notebook"/>
+                <param name="run_it" type="boolean" truevalue="true" falsevalue="false" label="Execute notebook and return a new one."
+                    help="This option is useful in workflows when you just want to execute a notebook and not dive into the webfrontend."/>
+            </when>
```

Comment:
The parameters from this branch of the conditional seem unused in the `<command>`.

Reply:
They are used in the else branch

Reply:
> They are used in the else branch

Yep, after my review :wink: